linera-core/src/worker.rs

```diff
@@ -834,9 +835,8 @@ where
         pending_blobs: &BTreeMap<BlobId, HashedBlob>,
     ) -> Result<Vec<HashedBlob>, WorkerError> {
         let mut blobs = Vec::new();
-        let mut recent_blobs = self.recent_hashed_blobs.lock().await;
         for blob_id in blob_ids {
-            if let Some(blob) = recent_blobs.get(&blob_id) {
+            if let Some(blob) = self.recent_hashed_blobs.get(&blob_id).await {
```

Comment:
Optional: This looks like a good opportunity to add a `try_get_many` method to the cache:

```rust
/// Tries to retrieve many values from the cache.
///
/// Returns one collection with the values found, and another collection with the keys that
/// aren't present in the cache.
pub async fn try_get_many::<FoundCollection, NotFoundCollection>(
    &self,
    keys: impl IntoIterator<Item = K>,
) -> (FoundCollection, NotFoundCollection)
where
    FoundCollection: FromIterator<(K, V)>,
    NotFoundCollection: FromIterator<K>,
{
    let mut cache = self.cache.lock().await;
    let (found_keys, not_found_keys) = keys.into_iter().partition(|key| cache.contains(key));

    let found_pairs = found_keys
        .into_iter()
        .map(|key| {
            let value = cache
                .get(key)
                .expect("Key should be in cache after the partitioning above");
            (key, value)
        })
        .collect();

    (found_pairs, not_found_keys)
}
```

This would also get the lock once, optimizing a bit.

Reply:
Nice! Will do